src/test/groovy/com/gooddata/executeafm/afm/OverPeriodDateAttributeTest.groovy

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (C) 2007-2018, GoodData(R) Corporation. All rights reserved.
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE.txt file in the root directory of this source tree.
+ */
+package com.gooddata.executeafm.afm
+
+import com.gooddata.executeafm.UriObjQualifier
+import nl.jqno.equalsverifier.EqualsVerifier
+import org.apache.commons.lang3.SerializationUtils
+import spock.lang.Specification
+
+import static com.gooddata.util.ResourceUtils.readObjectFromResource
+import static net.javacrumbs.jsonunit.JsonMatchers.jsonEquals
+import static net.javacrumbs.jsonunit.core.util.ResourceUtils.resource
+import static spock.util.matcher.HamcrestSupport.that
+
+class OverPeriodDateAttributeTest extends Specification {
+
+    private static final String ATTRIBUTE_JSON = "executeafm/afm/overPeriodDateAttribute.json"
+
+    def "should serialize"() {
+        expect:
+        OverPeriodDateAttribute dateAttribute = new OverPeriodDateAttribute(
+                new UriObjQualifier('/gdc/md/projectId/obj/1'), 2)
+        that dateAttribute, jsonEquals(resource(ATTRIBUTE_JSON))
+    }
+
+    def "should deserialize"() {
+        when:
+        OverPeriodDateAttribute dateAttribute = readObjectFromResource("/$ATTRIBUTE_JSON", OverPeriodDateAttribute)
+
+        then:
+        dateAttribute.attribute.uri == '/gdc/md/projectId/obj/1'
+        dateAttribute.periodsAgo == 2
+    }
+
+    def "test serializable"() {
+        given:
+        OverPeriodDateAttribute dateAttribute = readObjectFromResource("/$ATTRIBUTE_JSON", OverPeriodDateAttribute)
+        OverPeriodDateAttribute deserialized = SerializationUtils.roundtrip(dateAttribute)
+
+        expect:
+        that deserialized, jsonEquals(dateAttribute)
+    }
+
+    def "should throw when attribute is null"() {
+        when:
+        new OverPeriodDateAttribute(null, 2)
+        then:
+        thrown(NullPointerException)
+    }
+
+    def "should throw when periodsAgo is null"() {
+        when:
+        new OverPeriodDateAttribute(new UriObjQualifier('/gdc/md/projectId/obj/1'), null)
```

Comment:
consider to merge with previosu test, testing all 4 possibilities using where ;-)

Reply:
Fixed. In other tests as well.